docs/formatter.md

```diff
@@ -225,6 +225,81 @@ def f(x):
 [literal blocks]: https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#literal-blocks
 [`code-block` and `sourcecode` directives]: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block
 
+## F-string formatting
+
+_Stabilized in 2025 style with Ruff 0.9.0_
+
+By default, Ruff formats f-string expressions. This is a [known deviation](formatter/black.md#f-strings)
+from Black, which does not format f-strings.
+
+Ruff employs several heuristics to determine how an f-string should be formatted, including the
+following considerations:
+
+* The quote style for both outer and nested f-strings
+* Whether to break the f-string into multiple lines
+* When to skip formatting the f-string altogether
+
+### Quotes
+
+Ruff will use the configured quote style for the f-string expression unless doing so would result in
+invalid syntax for the target Python version. In such cases, Ruff will avoid changing the
+quote style.
+
+For example, if a self-documenting f-string contains a string literal with the configured quote
+style, Ruff cannot use the same quote style for the f-string itself when targeting Python versions
+earlier than 3.12:
+
+```python
+f'{10 + len("hello")=}'
+# This f-string cannot be formatted as follows when targeting Python < 3.12
+f"{10 + len("hello")=}"
+```
+
+For nested f-strings, Ruff alternates quote styles, starting with the configured style for the
+outermost f-string. If this would result in an invalid syntax, Ruff avoids changing the quote style.
+For example, consider the following f-string:
+
+```python
+f"outer f-string {f"nested f-string {f"another nested f-string"} end"} end"
+```
+
+Ruff formats it as:
+
+```python
+f"outer f-string {f'nested f-string {f"another nested f-string"} end'} end"
+```
+
+### Line breaks
+
+Starting with Python 3.12 ([PEP 701](https://peps.python.org/pep-0701/)), f-string expressions can
+span multiple lines. Ruff adopts a similar heuristic as [Prettier](https://prettier.io/docs/en/next/rationale.html#template-literals)
+for when to introduce a line break in an f-string expression: it introduces line breaks only
+if the original f-string expression already contains them.
+
+For example, the following code:
+
```

Comment:
I'm leaning towards motivating this closer to what Prettier wrote

> Template literals can contain interpolations. Deciding whether it's appropriate to insert a linebreak within an interpolation unfortunately depends on the semantic content of the template - for example, introducing a linebreak in the middle of a natural-language sentence is usually undesirable. Since Prettier doesn't have enough information to make this decision itself, it uses a heuristic similar to that used for objects: it will only split an interpolation expression across multiple lines if there was already a linebreak within that interpolation.

Reply:
I've provided a link to the relevant section in Prettier which is why I thought not to add the same detail here and keep it short.

Reply:
Including a link to the source of our inspiration is great but I don't expect many users to click it and one important aspect of the style guide section is that we motivate our formatting. That's why I'd still prefer at least a few words why we trade consistency with flexibility.